### PR TITLE
State Machine

### PR DESCRIPTION
**Pull request by NikhilPeri**
actually start working on the state machine for the rocket

### CODE REVIEW

**Comment by NikhilPeri** [comment]
you have to rebase and fix merge conflicts on this before merging.
```
git rebase master
<fix all merge conflicts marked with '<<<<<' >
git rebase  --continue
git push -f
```

File: app/rocket.py [hunk 1]
```diff
@@ -40,63 +41,62 @@ def __init__(self):
...
 
 
     def powered_state(self):
         if self.timer_init == True:
             self.starttime = time.time()
             self.timer_init = False
-        #TODO: read IMU  
-        #TODO: z_accel <-9.81 OR t > 3s
+ 
         if (self.kinetics.acceleration()['z'] < -9.81) or ((time.time() - self.starttime) > 3):
```
Comment (buzuloiu) [inline app/rocket.py]: Why 3 seconds?

File: app/rocket.py [hunk 2]
```diff
@@ -40,63 +41,62 @@ def __init__(self):
...
-                #TODO: deploy brakes
-                Rocket.deploy_brakes()
 
-            elif self.kinetics.apogee() < target:
-                #TODO: retract brakes
-                Rocket.retract_brakes()
+    def coast_state(self):
+        
+        target = 10000
+        if self.kinetics.apogee() > target:
```
Comment (buzuloiu) [inline app/rocket.py]: Not sure if this is how we should implement the actual control during flight.

File: test/test_rocket.py [hunk 3]
```diff
@@ -102,12 +102,9 @@ def bad_transitions():
...
     with pytest.raises(transitions.core.MachineError) as e_info:
         trans()
-            
+         
     print(e_info.value)
-    #try:
-    #    transition()
-    #except Exception as e: 
-    #    print(e)
+    print('Current state: ' + RockIt.state + '\n')
```
Comment (buzuloiu) [inline test/test_rocket.py]: A+ naming

File: app/rocket.py (hunk 1 above)
Comment (NikhilPeri) [reply app/rocket.py]: incase the accelerometer dies, we can still track state based on the simulation data

**Pull request merged**